src/features/profile/pages/ChangeEmail/ChangeEmail.native.test.tsx

```diff
@@ -43,155 +24,241 @@ jest.mock('ui/components/snackBar/SnackBarContext', () => ({
-    expect(analytics.logErrorSavingNewEmail).toHaveBeenCalledWith(
-      CHANGE_EMAIL_ERROR_CODE.INVALID_PASSWORD
-    )
-  })
+    it('should show the generic error message', async () => {
+      const { getByPlaceholderText, getByLabelText } = renderChangeEmail()
 
-  it('should show the generic error message if the API call returns an attempts limit error', async () => {
-    mockUseMutationError(CHANGE_EMAIL_ERROR_CODE.EMAIL_UPDATE_ATTEMPTS_LIMIT)
+      await fillInputs(getByPlaceholderText, {})
 
-    const { getByPlaceholderText, getByLabelText } = render(<ChangeEmail />)
-    const submitButton = getByLabelText('Enregistrer les modifications')
-    await act(async () => {
-      const emailInput = getByPlaceholderText('tonadresse@email.com')
-      fireEvent.changeText(emailInput, 'tonadresse@email.com')
-    })
-    await act(async () => {
-      const passwordInput = getByPlaceholderText('Ton mot de passe')
-      fireEvent.changeText(passwordInput, 'password>=12')
-    })
+      await submitForm(getByLabelText)
 
-    await act(async () => {
-      fireEvent.press(submitButton)
+      expect(mockShowErrorSnackBar).toBeCalledWith({
+        message: `Une erreur s’est produite pendant la modification de ton e-mail. Réessaie plus tard.`,
+        timeout: SNACK_BAR_TIME_OUT,
+      })
     })
 
-    expect(navigate).not.toBeCalled()
-    expect(mockShowErrorSnackBar).toBeCalledWith({
-      message: `Une erreur s’est produite pendant la modification de ton e-mail. Réessaie plus tard.`,
-      timeout: 5000,
+    it('should log analytics', async () => {
+      const { getByPlaceholderText, getByLabelText } = renderChangeEmail()
+
+      await fillInputs(getByPlaceholderText, {})
+
+      await submitForm(getByLabelText)
+
+      expect(analytics.logErrorSavingNewEmail).toHaveBeenCalledWith(
+        CHANGE_EMAIL_ERROR_CODE.EMAIL_UPDATE_ATTEMPTS_LIMIT
+      )
     })
-    expect(analytics.logErrorSavingNewEmail).toHaveBeenCalledWith(
-      CHANGE_EMAIL_ERROR_CODE.EMAIL_UPDATE_ATTEMPTS_LIMIT
-    )
   })
 })
+
+// eslint-disable-next-line local-rules/no-react-query-provider-hoc
+const renderChangeEmail = () => render(reactQueryProviderHOC(<ChangeEmail />))
+
+const fillInputs = async (
+  getByPlaceholderText: (predicate: string) => ReactTestInstance,
+  { email, password }: { email?: string; password?: string }
+) => {
+  const passwordInput = getByPlaceholderText('Ton mot de passe')
+  fireEvent.changeText(passwordInput, password ?? 'password>=12')
+
+  const emailInput = getByPlaceholderText('tonadresse@email.com')
+  fireEvent.changeText(emailInput, email ?? 'tonadresse@email.com')
+
+  // To avoid CI flakiness
+  await superFlushWithAct()
+}
+
+const submitForm = async (getByLabelText: (predicate: string) => ReactTestInstance) => {
+  const submitButton = getByLabelText('Enregistrer les modifications')
+  fireEvent.press(submitButton)
+
+  // To avoid CI flakiness
+  await superFlushWithAct()
+}
+
```

Comment:
Aujourd'hui j'ai eu une formation sur les nouveautés de react native testing library et il semblerait qu'on puisse utiliser "screen"  pour faire ça de manière plus joli (sans utiliser getByLabelText)
D'après la [doc](https://callstack.github.io/react-native-testing-library/docs/api/#screen) :
"Hold the value of latest render call for easier access to query and other functions returned by [render](https://callstack.github.io/react-native-testing-library/docs/api/#render)."

```jsx
import { screen } from 'test/utils'

const submitForm = async () => {
  const submitButton = screen.getByLabelText('Enregistrer les modifications')
  fireEvent.press(submitButton)

  // To avoid CI flakiness
  await superFlushWithAct()
}
```